plugins/catalog-react/src/filters.ts

```diff
@@ -188,3 +188,16 @@ export class EntityErrorFilter implements EntityFilter {
     return error !== undefined && this.value === error;
   }
 }
+
+/**
+ * Filters entities based on the metadata category (status, domain, capabilities etc) provided.
+ * @public
+ */
+export class EntityGenericFilter implements EntityFilter {
+  constructor(readonly values: string[], readonly category: string) {}
+  filterEntity(entity: Entity): boolean {
+    return this.values.every(v =>
+      ((entity.metadata[this.category] as string[]) ?? []).includes(v),
```

Comment:
I'm thinking it makes sense for this to be able to filter based on any field in the entity, not just metadata. So you'd give it something like `'metadata.status'`, or `'spec.capabilities'`, etc. Or we could keep it focused on metadata and call it `EntityMetadataFieldPicker`.

Is there also an assumption here that the metadata field is always `string[]`? It's not really a safe assumption and we'd need to deal with different forms of matching here I think

Reply:
Yup, it makes sense to filter based on any field in the entity.